src/Application/Economy/Commands/BalanceTransactionCommand.cs

```diff
@@ -0,0 +1,31 @@
+using Application.Services;
+using Application.Users.Commands;
+using Domain.Aggregates;
+using Domain.Abstractions;
+using Microsoft.EntityFrameworkCore;
+using FluentValidation;
+using MediatR;
+namespace Application.Economy.Commands;
+
+
+public sealed record BalanceTransactionCommand(User sender, User reciver, decimal amount) : IRequest<bool>;
```

Comment:
I think, i made a mistake, it will be better if we pass UserId-s here instead of users. because fetching the users from the database in the caller will not be very clean.

we would rather: fetch the users from the db inside the executor.

ps: add async validators inside the "async" ruleset.
```cs
// this needs to be in a separate ruleset, because ASP.NET does not support async validation, BUT our ValidationBehaaviour.cs does support async validation and it validates all rulesets. please look at it

// DI service injected in the constructor
public CommandValidator(IAppDbContext db)
{
  RuleSet("async", () => {
    RuleFor(x => x.SenderId)
      .MustAsync(async (id, ct) => {
        var usersWithId = await db.Users.Count(...); 
        return usersWithId == 1;
      });
  });
}
```